src/lib.rs

```diff
@@ -15,13 +15,15 @@
 #![cfg_attr(target_arch = "riscv64", feature(offset_of))]
 #![feature(allocator_api)]
 #![feature(asm_const)]
+#![feature(associated_type_defaults)]
 #![feature(exposed_provenance)]
 #![feature(linked_list_cursors)]
 #![feature(maybe_uninit_slice)]
 #![feature(naked_functions)]
 #![feature(new_uninit)]
 #![feature(noop_waker)]
 #![feature(pointer_is_aligned)]
+#![feature(ptr_metadata)]
```

Comment:
Can we avoid depending on these unstable features? We would like to decrease the dependence on unstable features if possible.